metrics/web/test/project_groups/presentation/widgets/project_group_dialog_test.dart

```diff
@@ -240,6 +239,155 @@ void main() {
+            projectGroupsNotifier: projectGroupsNotifier,
+          ));
+        });
+
+        final groupNameTextFormFieldFinder = find.ancestor(
+          of: find.text(ProjectGroupsStrings.nameYourGroup),
+          matching: find.byType(TextFormField),
+        );
+
+        await tester.enterText(groupNameTextFormFieldFinder, 'some group name');
+        await tester.pump();
+
+        expect(find.byType(MetricsPositiveButton), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the inactive create group button when the group name is not valid",
+      (tester) async {
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(_ProjectGroupDialogTestbed(
+            strategy: strategy,
+          ));
+        });
+
+        final groupNameTextFormFieldFinder = find.ancestor(
+          of: find.text(ProjectGroupsStrings.nameYourGroup),
+          matching: find.byType(TextFormField),
+        );
+
+        await tester.enterText(groupNameTextFormFieldFinder, '');
+        await tester.pumpAndSettle();
+
+        expect(find.byType(MetricsInactiveButton), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the inactive create group button when the number of selected projects is > 20",
+      (tester) async {
+        final List<String> selectedIds = [];
+        for (int i = 0; i <= 21; ++i) {
+          selectedIds.add("some value");
+        }
+        final projectDialogViewModel = ProjectGroupDialogViewModel(
+          id: "id",
+          name: "name",
+          selectedProjectIds: UnmodifiableListView<String>(selectedIds),
+        );
+
+        when(projectGroupsNotifier.projectGroupDialogViewModel)
+            .thenReturn(projectDialogViewModel);
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(_ProjectGroupDialogTestbed(
+            strategy: strategy,
+            projectGroupsNotifier: projectGroupsNotifier,
+          ));
+        });
+
+        expect(find.byType(MetricsInactiveButton), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the inactive create group button when there are no selected projects",
+      (tester) async {
+        final projectDialogViewModel = ProjectGroupDialogViewModel(
+          id: "id",
+          name: "name",
+          selectedProjectIds: UnmodifiableListView<String>([]),
+        );
+
+        when(projectGroupsNotifier.projectGroupDialogViewModel)
+            .thenReturn(projectDialogViewModel);
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(_ProjectGroupDialogTestbed(
+            strategy: strategy,
+            projectGroupsNotifier: projectGroupsNotifier,
+          ));
+        });
+
+        expect(find.byType(MetricsInactiveButton), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the inactive create group button when the group name is not valid even if the number of selected projects is valid",
```

Comment:
```suggestion
      "displays the inactive create group button when the group name is not valid and the number of selected projects is valid",
```
maybe we rephrase it like this to simplify reading? I think "even" might be a bit not formal or emotional?